File: cesta/IvanRC-Cesta/IvanRC-Cesta-Java/src/main/java/FactoriesCesta/FactoryCesta.java
package FactoriesCesta;

import ModelsCesta.Cesta;
import ModelsCesta.ListaCesta;

import java.time.LocalDate;

public class FactoryCesta {

    public FactoryCesta(){}
    private static FactoryCesta instance = null;

    public static FactoryCesta getInstance(){
        if(instance == null){
            instance = new FactoryCesta();
        }
        return instance;
    }
    private static int contadorCesta = 1;
    private int nextId(){
        return contadorCesta++;
    }

    /**
     * función que srive para crear una cesta random
     * @return la cesta creada aleatoriamente
     */
    public Cesta crearCestaRandom() {
        String[] fechas = {"2023-12-03", "2003-02-23", "2023-12-03", "2023-12-03", "2023-12-03", "2023-12-03", "2023-12-03"};
        return new Cesta(nextId(), LocalDate.parse(fechas[(int) (Math.random()*fechas.length)]), FactoryUsuario.getInstance().crearUsuarioRandom(), new ListaCesta());
    }
}

File: cesta/IvanRC-Cesta/IvanRC-Cesta-Kotlin/src/main/kotlin/FactoriesCestaDeCompra/FactoryCesta.kt
package FactoriesCestaDeCompra

import ModelsCestaDeCompra.Cesta
import ModelsCestaDeCompra.ListaCesta
import java.time.LocalDate

class FactoryCesta {

    companion object{
        private var instance: FactoryCesta? = null

        fun getInstance(): FactoryCesta?{
            if(instance == null){
                instance = FactoryCesta()
            }
            return instance
        }
        var contadorCestas = 1
        fun nextId():Int{
            return contadorCestas++
        }
    }

    /**
     * función que srive para crear una cesta random
     * @return la cesta creada aleatoriamente
     */
    fun crearCestaRandom(): Cesta {
        val fechas = arrayOf("2023-12-03", "2003-02-23", "2023-12-03", "2023-12-03", "2023-12-03", "2023-12-03", "2023-12-03")
        return Cesta(nextId(), LocalDate.parse(fechas.random()), FactoryUsuario.getInstance()!!.crearUsuarioRandom(), ListaCesta())
    }
}